### PR TITLE
feat: update label with VibeComponent

### PR DESCRIPTION
**Pull request by jes14**
make <Label> a VibeComponent and add ref forwarding (https://github.com/mondaycom/monday-ui-react-core/issues/1588)

### CODE REVIEW

File: src/components/Label/Label.tsx [hunk 1]
```diff
@@ -28,75 +29,82 @@ interface LabelProps extends VibeComponentProps {
...
+      "data-testid": dataTestId,
+      onClick
     },
-    labelRef
-  );
+    ref
+  ) => {
+    const overrideClassName = backwardCompatibilityForProperties([className, wrapperClassName]) as string;
+    const isClickable = Boolean(onClick);
+    const mergedRef = useMergeRefs({ refs: [ref] });
```
Comment (SergeyRoyt) [inline src/components/Label/Label.tsx:54]: `labelRef` should also be among mergedRefs

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
One issue to fix :)

**Comment by jes14** [comment]
Hi @SergeyRoyt 
Thanks a lot for your reviews. Just updated it.

**Review by SergeyRoyt — APPROVED** [review]
Looks great now!
Thanks a lot for your contribution! ❤️

**Pull request merged**